shop/forms/checkout.py

```diff
@@ -69,7 +69,7 @@ def __init__(self, initial=None, instance=None, *args, **kwargs):
     def form_factory(cls, request, data, cart):
         customer_form = cls(data=data, instance=request.customer.user)
         if customer_form.is_valid():
-            customer_form.instance.recognize_as_guest(request, commit=False)
+            customer_form.instance.customer.recognize_as_guest(request, commit=False)
```

Comment:
I have no objections against using `request.customer`. I fixed it in a hurry because it was blocking me ;) I'll update the PR later.